tools/src/main/java/org/neo4j/tools/applytx/InteractiveApplyTransactions.java

```diff
@@ -0,0 +1,443 @@
+    {
+        DependencyResolver resolver = toDb.getDependencyResolver();
+        BatchingTransactionRepresentationStoreApplier applier = new BatchingTransactionRepresentationStoreApplier(
+                resolver.resolveDependency( IndexingService.class ),
+                resolver.resolveDependency( LabelScanStore.class ),
+                resolver.resolveDependency( NeoStoreProvider.class ).evaluate(),
+                resolver.resolveDependency( CacheAccessBackDoor.class ),
+                resolver.resolveDependency( LockService.class ),
+                resolver.resolveDependency( LegacyIndexApplierLookup.class ),
+                resolver.resolveDependency( IndexConfigStore.class ),
+                resolver.resolveDependency( KernelHealth.class ),
+                resolver.resolveDependency( IdOrderingQueue.class ) );
+        TransactionRepresentationCommitProcess commitProcess =
+                new TransactionRepresentationCommitProcess(
+                        resolver.resolveDependency( LogicalTransactionStore.class ),
+                        resolver.resolveDependency( NeoStoreProvider.class ).evaluate(),
+                        applier,
+                        resolver.resolveDependency( IndexUpdatesValidator.class ) );
+        LifeSupport life = new LifeSupport();
+        DefaultFileSystemAbstraction fileSystem = new DefaultFileSystemAbstraction();
+        try ( PageCache pageCache = StandalonePageCacheFactory.createPageCache( fileSystem ) )
+        {
+            LogicalTransactionStore source = life.add( new ReadOnlyTransactionStore( fileSystem, fromPath,
+                    new Monitors(), resolver.resolveDependency( KernelHealth.class ) ) );
+            life.start();
+            // Some progress if there are more than a couple of transactions to apply
+            ProgressListener progress = toTxInclusive - fromTxExclusive >= 100 ?
+                                        textual( System.out ).singlePart( "Application progress",
+                                                toTxInclusive - fromTxExclusive ) :
+                                        ProgressListener.NONE;
+            try ( IOCursor<CommittedTransactionRepresentation> cursor = source.getTransactions( fromTxExclusive + 1 ) )
+            {
+                while ( cursor.next() )
+                {
+                    try ( LockGroup locks = new LockGroup() )
+                    {
+                        commitProcess.commit( cursor.get().getTransactionRepresentation(), locks,
+                                CommitEvent.NULL, RECOVERY );
+                        progress.add( 1 );
+                    }
+                    catch ( final Exception e )
+                    {
+                        Exceptions.withMessage( e,
+                                "[ERROR applying transaction " + cursor.get().getCommitEntry().getTxId() +
+                                "]: " + e.getMessage() );
```

Comment:
Changing the exception message like this is pretty insane. Why do we need to do that? Why not simply throw a new exception?